operator/bucketcontroller/connector.go

```diff
@@ -41,48 +38,29 @@ func (c *bucketConnector) Connect(ctx context.Context, mg resource.Managed) (man
 		return &NoopClient{}, nil
 	}
 
-	pctx := &connectContext{
-		GenericConnectContext: controllerutil.GenericConnectContext{
-			Context:            ctx,
-			ProviderConfigName: bucket.GetProviderConfigName(),
-		},
-		EndpointURL: bucket.Spec.ForProvider.EndpointURL,
+	exo, err := pipelineutil.OpenExoscaleClient(ctx, c.Kube, bucket.GetProviderConfigName())
```

Comment:
I expect this function to be called very often in the near future with all the services we will support from Exoscale. Maybe it is a good idea to cash the client for a short amount of time?

Reply:
Not worth the complexity IMO. controller-runtime already caches objects when first retrieving (e.g. Secrets, providerconfig). Creating the client itself is inexpensive as well, as there's no "login" process. The heavyweight operations are still the API requests and those can hardly be cached.